### PR TITLE
Add cli based example and test

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
I finally did it... Here is a test that can be run (and add to coverage) that executes the CLI and can generate md formatted output that can be used for examples.

The idea is to actually go through the steps of the example in an automated way and use that for showing the output.
It is a bit repetitive but I think easy enough to read thanks to the descriptions.

Some assumptions are made about the naming scheme of the modules which is largely due to hooking the the coverage reports.

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
ping @leandrolanzieri and/or @Teufelchen1

File: tests/test_example_usage.py [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+    ct.run_step(
+        description="Let's just create a board-info list with some features...\n"
+        "We can override the default args to just echo directly.",
+        cmd="inet-nm-update-from-os",
+        args=[
+            ".",
+            "-i",
+            '"echo board_1 board_2"',
+            "-f",
+            '"echo feature_common feature_${BOARD}',
```
Comment (Teufelchen1) [inline tests/test_example_usage.py:104]: Typo: Missing " at end of line.

File: tests/test_example_usage.py [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+        "missing boards.",
+        cmd="inet-nm-check",
+        args=["-m"],
+    )
+    assert "board_1" in ret
+    assert "board_2" in ret
+    assert "board_3" in ret
+
+    ret = ct.run_step(
+        description="Maybe I only want boards with a feature, let's see what "
```
Comment (Teufelchen1) [inline tests/test_example_usage.py:169]: Did you mean:  `... with a specific feature, ..`

File: tests/test_example_usage.py [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+    @staticmethod
+    def _cmd_to_coverage(cmd):
+        cmd = cmd.replace("inet-nm-", "inet_nm.cli_")
+        cmd = cmd.replace("-", "_")
+        return f"coverage run -m {cmd}"
+
+    def run_step(
+        self,
+        cmd,
+        args=None,
```
Comment (Teufelchen1) [inline tests/test_example_usage.py:31]: Why not default to `[]`? Saves the `.. or []` down the line?

Comment (MrKevinWeiss) [reply tests/test_example_usage.py:31]: [It goes against the python recommendations](https://docs.python.org/3/reference/compound_stmts.html#function-definitions)

> Default parameter values are evaluated from left to right when the function definition is executed. This means that the expression is evaluated once, when the function is defined, and that the same “pre-computed” value is used for each call. This is especially important to understand when a default parameter value is a mutable object, such as a list or a dictionary: if the function modifies the object (e.g. by appending an item to a list), the default parameter value is in effect modified. This is generally not what was intended. A way around this is to use None as the default, and explicitly test for it in the body of the function, e.g.:
```py
def whats_on_the_telly(penguin=None):
    if penguin is None:
        penguin = []
    penguin.append("property of the zoo")
    return penguin
```

Comment (Teufelchen1) [reply tests/test_example_usage.py:31]: 🤯 wow! I Never knew! Makes me wonder how much of my Code only works by accident...

**Comment by MrKevinWeiss** [comment]
Thank you for the review,  I have added the changes as well as a design decisions section explaining the cli naming conventions.

**Pull request merged**